os/drivers/audio/audio_null.c

```diff
@@ -1000,8 +1009,24 @@ static int null_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned long
 	case AUDIOIOC_STOPPROCESS: {
 		auddbg("set stop process!!\n");
 #ifdef CONFIG_AUDIO_PROCESSING_FEATURES
-		priv->speech_state = AUDIO_NULL_SPEECH_STATE_NONE;
-		priv->process_terminate = true;
+		if ((uint8_t)arg == AUDIO_SD_KEYWORD_DETECT) {
```

Comment:
This is acceptable for null driver, but our another target board should use one thread.
So, when driver_stop or process_stop called, it should check priv->running & priv->kd, epd.
currently this is what I want.